testing/maas_billing_tests_independent/tests/README.MD

```diff
@@ -0,0 +1,350 @@
+pytest -q testing/maas_billing_tests_independent/tests/test_gateway_endpoints.py::test_chat_completion_works \
+  --html="$OUT/chat.html" --self-contained-html \
+  --junitxml="$OUT/chat.xml"
+
+---
+
+## 5) What each test does (cheat-sheet)
+
+- `tests/test_tokens.py` – mints a short-lived MaaS JWT from your OC token.
+- `tests/test_models_user.py` – lists models; asserts your `MODEL_NAME` exists.
+- `tests/test_gateway_endpoints.py` – discovers the model’s **URL** via `/v1/models`
+  then calls **`<model-url>/v1/chat/completions`** and expects a normal reply.
+- `tests/test_quota_global.py` – sends **N = RATE_LIMIT_BURST_FREE + 5** quick calls with
+  low tokens; expects ≥ `RATE_LIMIT_BURST_FREE` successes and then **429** (request-rate limiter).
+- `tests/test_quota_per_user.py` – runs the same quick burst for Free & Premium; asserts
+  **Premium ≥ Free** (Premium is not more restricted).
+- `tests/test_token_ratelimit.py` (optional) – cranks up token usage to hit the token-rate
+  limiter and prints usage headers:
+  `x-odhu-usage-input-tokens`, `x-odhu-usage-output-tokens`, `x-odhu-usage-total-tokens`.
+- `tests/test_usage_logs.py` (optional) – smoke call + probe of the Usage API endpoint.
+
+---
+
+## 6) PowerShell equivalents (Windows)
+
+```powershell
+# venv
+python -m venv .venv
+. .\.venv\Scripts\Activate
+pip install -r requirements.txt
+# (optional) pip install pytest-html
+
+# login (role of your choice)
+oc login https://api.<cluster>:6443 --token '<token>'
+
+# resolve Route host (avoid $HOST which is reserved in PS)
+$CLUSTER_DOMAIN = oc get ingresses.config.openshift.io cluster -o jsonpath='{.spec.domain}'
+$ROUTE_HOST = "maas-api.$CLUSTER_DOMAIN"
+
+# MaaS API base URL (route-based)
+$env:MAAS_API_BASE_URL = "https://$ROUTE_HOST"
+$env:USAGE_API_BASE    = $env:MAAS_API_BASE_URL
+
+$env:FREE_OC_TOKEN     = oc whoami -t
+$env:MODEL_NAME        = "<model-id>"        # from /v1/models
+
+# request-rate bursts
+$env:RATE_LIMIT_BURST_FREE    = "5"          # put your Free limit here
+$env:RATE_LIMIT_BURST_PREMIUM = "20"         # put your Premium limit here
+
+# keep token usage low in request-rate tests
+$env:TOKENS_PER_CALL_SMALL = "16"
+$env:BURST_SLEEP = "0.05"
+
+# run
+pytest -q tests/test_tokens.py::test_minted_token_is_jwt
+pytest -q tests/test_models_user.py
+pytest -q tests/test_gateway_endpoints.py::test_chat_completion_works
+pytest -q tests/test_quota_global.py::test_rate_limit_burst
+
+# report (timestamped artifacts folder)
+$REP_ROOT = "testing\maas_billing_tests_independent\artifacts"
+$TS = Get-Date -Format "yyyy-MM-dd_HH-mm-ss"
+$OUT = "$REP_ROOT\$TS"
+New-Item -ItemType Directory -Force -Path $OUT | Out-Null
+
+pytest -q testing/maas_billing_tests_independent/tests `
+  --html="$OUT\maas-test-report.html" --self-contained-html `
+  --junitxml="$OUT\maas-test-report.xml"
+
+Write-Host "Reports saved in: $OUT"
+```
+
+> In PowerShell, **do not** use `$HOST`; use `$ROUTE_HOST`.
+
+---
+
+## 8) TL;DR (copy-paste)
```

Comment:
Are these the consolidated steps required to run testing? can we create a simple script instead? or `make e2e`?